GVFS/GVFS.Common/GVFSPlatform.cs

```diff
@@ -25,6 +26,7 @@ public GVFSPlatform(string executableExtension, string installerExtension, Under
 
         public GVFSPlatformConstants Constants { get; }
         public UnderConstructionFlags UnderConstruction { get; }
+        public DiskLayoutVersion DiskLayout { get; }
```

Comment:
What are your thoughts on moving this into `IDiskLayoutUpgrade`?

Although it's not *strictly* related to the upgraders, perhaps they're close enough it's worth moving them?  (It might make the relationship between these versions and the upgraders a bit clearer).

/cc: @nickgra for his thoughts as well

Reply:
Do you mean IDiskLayoutUpgradeData?  I didn't see IDiskLayoutUpgrade.

I must admit I'm not super familiar the upgrade relationship myself, but since this just reads repo metadata I'm not sure I'd expect it with upgrade.

Reply:
> Do you mean IDiskLayoutUpgradeData?

Yep sorry about that.

> but since this just reads repo metadata I'm not sure I'd expect it with upgrade.

The relationship I was thinking about was that the disk layout version is what drives the upgraders.

See this comment for `CurrentMajorVersion`

```
// The major version should be bumped whenever there is an on-disk format change that requires a one-way upgrade.
```

Reply:
To be clear the proposed change would be to make:

    public interface IDiskLayoutUpgradeData
    {
        DiskLayoutUpgrade[] Upgrades { get; }
        DiskLayoutVersion Version { get; }
        bool TryParseLegacyDiskLayoutVersion(string dotGVFSPath, out int majorVersion);
    }

I can definitely do that if we think we should.   @wilbaker let me know your final verdict :-)

Reply:
I think the proposed change makes sense, the reasoning that the disk layout version drives the upgraders makes sense to me.

Reply:
@jeschu1 I think the proposed interface change to `IDiskLayoutUpgradeData` looks.